R/zchunk_L2112.ag_nonco2_IRR_MGMT.R

```diff
@@ -1,135 +1,193 @@
+      L2112.AGRBio
+
+    L2111.AWB_BCOC_EmissCoeff %>%
+      repeat_add_columns(tibble::tibble(level = c("hi", "lo"))) %>%
+      unite(AgProductionTechnology, AgProductionTechnology, level, sep = "_") ->
+      L2112.AWB_BCOC_EmissCoeff
+
+    L2111.nonghg_max_reduction %>%
+      repeat_add_columns(tibble::tibble(level = c("hi", "lo"))) %>%
+      unite(AgProductionTechnology, AgProductionTechnology, level, sep = "_") ->
+      L2112.nonghg_max_reduction
+
+    L2111.nonghg_steepness %>%
+      repeat_add_columns(tibble::tibble(level = c("hi", "lo"))) %>%
+      unite(AgProductionTechnology, AgProductionTechnology, level, sep = "_") ->
+      L2112.nonghg_steepness
+
+
+    # For the tables whose emissions are read as quantities rather than rates,
+    # disaggregate emissions on the basis of production.
+
+    # First calculate the total emissions for each region, supply sector, subsector,
+    # technology level in the most recent model base year. This total will latter be
+    # used to  calculate share weights.
+    L2012.AgProduction_ag_irr_mgmt %>%
+      filter(year == max(BASE_YEARS)) %>%
+      mutate(AgProductionTechnology_nolvl = gsub("_hi|_lo", "", AgProductionTechnology)) %>%
+      group_by(region, AgSupplySector, AgSupplySubsector, AgProductionTechnology_nolvl, year) %>%
+      summarise(total = sum(calOutputValue)) %>%
+      ungroup(.) ->
+      L2112.AgProduction_ag_irr_nomgmt_aggergate
+
+    # Now subset the table of agricultural production for the most recent model base year
+    # and format the data frame so that so that the AgProductionTechnology_level column
+    # will match the column in the aggeregated data frame.
+    L2012.AgProduction_ag_irr_mgmt %>%
+      filter(year == max(BASE_YEARS)) %>%
+      select(region, AgSupplySector, AgSupplySubsector, AgProductionTechnology_lvl = AgProductionTechnology, year, calOutputValue) %>%
+      mutate(AgProductionTechnology_nolvl = gsub("_hi|_lo", "", AgProductionTechnology_lvl)) ->
+      L2112.AgProduction_ag
+
+    # Calculate the share weights or the fraction of emissions for each region, sector, subsector,
+    # technology and level for the using the aggrated total emissions determined above.
+    # Eventually these emission shares will be used as a factor to disaggregate the emissions input.
+    L2112.AgProduction_ag %>%
+      left_join_error_no_match(L2112.AgProduction_ag_irr_nomgmt_aggergate,
+                               by = c("region", "AgSupplySector", "year", "AgSupplySubsector", c("AgProductionTechnology_nolvl"))) %>%
+      mutate(share_tech = calOutputValue / total) ->
+      L2112.AgProduction_ag_share
+
+
+    # Combine non agricultural waste burning emissions and agricultural waste burning
+    # emissions into a single data frame. Add management level information to production
+    # technology column.
+    L2111.AWBEmissions %>%
+      bind_rows(L2111.AGREmissions) %>%
+      repeat_add_columns(tibble::tibble(level = c("hi", "lo"))) %>%
```

Comment:
Here and throughout, this is OK, but you can also just write `tibble` (no namespace qualifier)